### PR TITLE
ruff: check docstring format

### PR DESCRIPTION
**Pull request by ahms5**
- add D100 - D107 to ruff and add remaining docs
- here are the things we need to fix https://circleci.com/gh/pyfar/pyfar/8710
- ChatGPT and I have added documentation where its required. Please also crosscheck the content.

### CODE REVIEW

**Review by f-brinkmann — CHANGES_REQUESTED** [review]
Thanks for the effort. I generarly like having this. In some cases you added documentation that shows up in the build docs:

- classes/orientations.py
- dsp/dsp.py
- utils.py

In those cases we should put in a little more effort. But maybe wait until we have the opinon from others. In case of utils.py a proper docstring is introduced in #591

File: pyfar/classes/orientations.py [hunk 1]
```diff
@@ -1,3 +1,4 @@
+"""This module contains the Orientations class. """
```
Comment (ahms5) [inline pyfar/classes/orientations.py:1]: shows up in doc

File: pyfar/dsp/__init__.py [hunk 2]
```diff
@@ -1,3 +1,5 @@
+"""Digital signal processing functions."""
```
Comment (ahms5) [inline pyfar/dsp/__init__.py:1]: shows up in doc

File: pyfar/io/__init__.py [hunk 3]
```diff
@@ -1,3 +1,7 @@
+"""
```
Comment (ahms5) [inline pyfar/io/__init__.py:1]: shows up in doc

**Review by ahms5 — COMMENTED** [review]
marked all parts of the document that are part of the document.

File: pyfar/classes/orientations.py (hunk 1 above)
Comment (ahms5) [reply pyfar/classes/orientations.py:1]: maybe we should shift the doc from the class to the toplevel

**Review by f-brinkmann — APPROVED** [review]

**Pull request merged**